spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -1691,11 +1691,14 @@ private void doPauseConsumerIfNecessary() {
 			if (!this.consumerPaused && (isPaused() || this.pausedForAsyncAcks)
 					|| this.pauseForPending) {
 
-				this.consumer.pause(this.consumer.assignment());
-				this.consumerPaused = true;
-				this.pauseForPending = false;
-				this.logger.debug(() -> "Paused consumption from: " + this.consumer.paused());
-				publishConsumerPausedEvent(this.consumer.assignment());
+				Collection<TopicPartition> assigned = getAssignedPartitions();
+				if (!CollectionUtils.isEmpty(assigned)) {
```

Comment:
I guess this one has to come together with an `import org.springframework.util.CollectionUtils;`